elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreHelper.java

```diff
@@ -195,4 +216,99 @@ private void resolveMetricMap(Set<Metric> metrics) {
         return entityProjection.getRelationships().stream()
                 .map(Relationship::getName).collect(Collectors.toCollection(LinkedHashSet::new));
     }
+
+    /**
+     * Validate the having clause before execution. Having clause is not as flexible as where clause,
+     * the fields in having clause must be either or these two:
+     * 1. A grouped by dimension in this query
+     * 2. An aggregated metric in this query
+     *
+     * All grouped by dimensions are defined in the entity bean, so the last entity class of a filter path
+     * must match entity class of the query.
+     *
+     * @param havingClause having clause generated from this query
+     */
+    private void validateHavingClause(FilterExpression havingClause) {
+        if (havingClause instanceof FilterPredicate) {
```

Comment:
This functions should be broken up or we'll get cyclomatic complexity lint errors